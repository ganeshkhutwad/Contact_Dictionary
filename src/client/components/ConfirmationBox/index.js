export { default as ConfirmationBox } from './ConfirmationBox';