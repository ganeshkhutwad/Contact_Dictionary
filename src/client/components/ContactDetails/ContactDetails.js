import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classnames from 'classnames';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Collapse from '@material-ui/core/Collapse';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import red from '@material-ui/core/colors/red';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ShareIcon from '@material-ui/icons/Share';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import MoreVertIcon from '@material-ui/icons/MoreVert';

const labelStyle = {
    width: '250px',
    display: 'inline-block'
};

const statusSection = {
    width: '100%',
    height: '50px',
    backgroundColor: 'green'
};

const styles = theme => ({
  card: {
    maxWidth: 600,
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  actions: {
    display: 'flex',
  },
  expand: {
    transform: 'rotate(0deg)',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
    marginLeft: 'auto',
    [theme.breakpoints.up('sm')]: {
      marginRight: -8,
    },
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {
    backgroundColor: red[500],
  },
  cardContent: {
      lineHeight: 2
  }
});

const ContactDetails = (props) => {
    const { classes } = props;

    return (
        <Card className={classes.card}>
        <CardHeader
            avatar={
            <Avatar aria-label="Recipe" className={classes.avatar}>
                GK
            </Avatar>
            }
            action={
            <IconButton>
                <MoreVertIcon />
            </IconButton>
            }
            title="Ganesh Khutwad"
            subheader="9763484283"
        />
        <CardContent>
            <div>
                <span style={labelStyle}>First Name</span>
                <span>Ganesh</span>
            </div>
            <div>
                <span style={labelStyle}>Last Name</span>
                <span>Khutwad</span>
            </div>
            <div>
                <span style={labelStyle}>Mobile Number</span>
                <span>9763484283</span>
            </div>
            <div>
                <span style={labelStyle}>Email address</span>
                <span>ganeshkhutwad30690@gmail.com</span>
            </div>
        </CardContent>
        <div style={statusSection}>
                <span>Active</span>
            </div>
        </Card>
    );
}

ContactDetails.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(ContactDetails);