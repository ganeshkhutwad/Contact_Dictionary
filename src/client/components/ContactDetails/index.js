export { default as ContactDetails } from './ContactDetails';