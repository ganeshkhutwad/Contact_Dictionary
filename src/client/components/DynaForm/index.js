export { default as DynaForm } from './DynaForm';
