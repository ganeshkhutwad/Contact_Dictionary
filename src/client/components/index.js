export { Header } from './Header';
export { ContactList } from './ContactList';
export { ContactDetails } from './ContactDetails';
export { AddButton } from './AddButton';
