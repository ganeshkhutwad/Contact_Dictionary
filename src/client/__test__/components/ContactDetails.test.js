/**
Test `AddButton` Presentational Component.
@author Ganesh Khutwad
 */
import React from 'react';
import { ContactDetails } from '../../components/ContactDetails';

const classes = { };
const list = { };

describe('`ContactDetails` Tests', () => {
    it('`ContactDetails` Snapshot', () => {
        const contactDetailsWrapper = shallow(<ContactDetails classes={classes} list={list} />);
        expect(contactDetailsWrapper).toMatchSnapshot();
    });
});
