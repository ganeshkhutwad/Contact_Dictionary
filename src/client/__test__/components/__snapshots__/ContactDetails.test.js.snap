// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`ContactDetails\` Tests \`ContactDetails\` Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ContactDetails)
    classes={Object {}}
    list={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "actions": "ContactDetails-actions-6",
        "active": "ContactDetails-active-3",
        "avatar": "ContactDetails-avatar-9",
        "card": "ContactDetails-card-1",
        "cardContent": "ContactDetails-cardContent-10",
        "expand": "ContactDetails-expand-7",
        "expandOpen": "ContactDetails-expandOpen-8",
        "inactive": "ContactDetails-inactive-4",
        "label": "ContactDetails-label-2",
        "media": "ContactDetails-media-5",
      },
      "list": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "actions": "ContactDetails-actions-6",
          "active": "ContactDetails-active-3",
          "avatar": "ContactDetails-avatar-9",
          "card": "ContactDetails-card-1",
          "cardContent": "ContactDetails-cardContent-10",
          "expand": "ContactDetails-expand-7",
          "expandOpen": "ContactDetails-expandOpen-8",
          "inactive": "ContactDetails-inactive-4",
          "label": "ContactDetails-label-2",
          "media": "ContactDetails-media-5",
        },
        "list": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
